helix-term/src/ui/editor.rs

```diff
@@ -1021,9 +1015,21 @@ impl EditorView {
             callback: None,
             on_next_key_callback: None,
         };
```

Comment:
Is it ok to unconditionally construct a context here? It doesn't look like it's mutative in any way--so was this just about avoiding unnecessary stack allocations before?

Reply:
I feel that this is a bit ugly and causes some code duplication.

Reply:
Sorry for the delay! I've been on vacation and trying not to program as much. I've got the itch again, though, so I'm back 🙂 

Do you mind elaborating more on what part is ugly/what part causes code duplication? If you're talking about the disjoint `if` statements below: I agree, but I tend to prefer duplication over a poor abstraction. I can't think of a good way to represent this without generalizing the concept of a timer which is a much bigger task to bite off.

With all of that said: please let me know if you have another preferred method of representing this! I'm not a regular contributor here so I'm happy to defer to those who are.